src/jdbc/cubrid/jdbc/jci/UConnection.java

```diff
@@ -1835,11 +1841,16 @@ UInputBuffer send_recv_msg() throws UJciException, IOException {
 
 	public boolean isValid(int timeout) throws SQLException {
 		boolean valid = !isClosed;
-
-                if (protoVersionIsAbove(PROTOCOL_V9)) {
-		    valid = BrokerHandler.isValid(CASIp, CASPort, processId, sessionId, timeout);
-		}
-		return valid;
+		try {
+			int status = BrokerHandler.statusBroker(CASIp, CASPort, processId, sessionId, timeout);
+			if (status == UConnection.FN_STATUS_NONE) {
+    				return false;
+    	    		}
+    		} catch (Exception e) {
+    			return false;
+    		}
```

Comment:
Protocol version checking seems to be missing. `protoVersionIsAbove (PROTOCOL_V9)`